src/Ledger/PPUp.lagda

```diff
@@ -92,7 +96,7 @@ data _⊢_⇀⦇_,PPUP⦈_ : PPUpdateEnv → PPUpdateState → Maybe Update →
   PPUpdateCurrent : let open PPUpdateEnv Γ in
     dom (pup ˢ) ⊆ dom (genDelegs ˢ)
     → All (isViableUpdate pparams) (range (pup ˢ))
-    → (slot + (2 * StabilityWindow)) <ˢ firstSlot (sucᵉ (epoch slot))
+    → (slot + (2 * StabilityWindow)) < firstSlot (sucᵉ (epoch slot))
```

Comment:
This single mod is the whole point of this PR and is the reason for adding the `HasStrictTotalOrder` type class.